src/Microsoft.TemplateEngine.Edge/Template/TemplateCreator.cs

```diff
@@ -302,14 +318,14 @@ public void ResolveUserParameters(ITemplate template, IParameterSet templatePara
                     }
                     else
                     {
-                        object? resolvedValue = template.Generator.ConvertParameterValueToType(_environmentSettings, paramFromTemplate, inputParam.Value, out bool valueResolutionError);
+                        object? resolvedValue = template.Generator.ConvertParameterValueToType(_environmentSettings, paramFromTemplate, inputParam.Value.ToString(), out bool valueResolutionError);
```

Comment:
does it make sense to do `ToString()` here? it looks like object should be passed instead.

Reply:
This is currently keepeing logic as is (previously the value was `string`, hence the `.ToString()` here).
It definitely would make sense to rework the internal logic of converting values to typed values - however I draw a refactoring line in sand here (as it's not this single method, but whole set of functions that all strongly relies on strings)